### PR TITLE
Fix ESP8266 build

### PR DESCRIPTION
**Pull request by seanshpark**
This change is to fix build for ESP8266 board with jerry_api_value_t changes.

JerryScript-DCO-1.0-Signed-off-by: SaeHie Park saehie.park@samsung.com

### CODE REVIEW

**Comment by seanshpark** [comment]
Though this is just a build fix, it fails to run in the target board for some reason that needs investigation. I'll work on this after this PR lands.

File: targets/esp8266/source/jerry_run.c [hunk 1]
```diff
@@ -45,7 +46,8 @@ int js_entry (const char *source_p, const size_t source_size)
   {
     if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
     {
-      ret_code = jerry_run ();
+      jerry_api_value_t res;
+      ret_code = jerry_run (&res);
```
Comment (LaszloLango) [inline targets/esp8266/source/jerry_run.c]: `jerry_api_release_value` should be called if jerry_run failes. The parameter of `jerry_run` is for error value not for result, so the name of the variable is misleading.

File: targets/esp8266/source/jerry_run.c [hunk 2]
```diff
@@ -45,7 +46,8 @@ int js_entry (const char *source_p, const size_t source_size)
   {
     if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
     {
-      ret_code = jerry_run ();
+      jerry_api_value_t res;
```
Comment (LaszloLango) [inline targets/esp8266/source/jerry_run.c]: It is not mandatory, but good practice to initialize `res` to `VOID` (`jerry_api_create_void_value`).

File: targets/esp8266/source/jerry_run.c (hunk 1 above)
Comment (seanshpark) [inline targets/esp8266/source/jerry_run.c]: got it. thanks.

**Comment by zherczeg** [comment]
LGTM

**Comment by LaszloLango** [comment]
LGTM

**Comment by LaszloLango** [comment]
@seanshpark, please rebase to the current master, so we can land your PR.

**Comment by seanshpark** [comment]
Done rebase, thank you.

**Comment by seanshpark** [comment]
CI fails, last merge is #1104. @zherczeg , can you please check ?

**Comment by zherczeg** [comment]
We have already spent a lot of time reproducing this issue, but no luck so far. It only appears on travis arm build. I submitted another patch, and it disappeared.

**Comment by seanshpark** [comment]
Thank you, rebase and merged.

**Pull request merged**